packages/desktop-client/src/components/mobile/rules/MobileRuleEditPage.tsx

```diff
@@ -77,6 +81,48 @@ export function MobileRuleEditPage() {
     navigate('/rules');
   };
 
+  const handleDelete = () => {
+    // Runtime guard to ensure id exists
+    if (!id || id === 'new') {
+      throw new Error('Cannot delete rule: invalid id');
+    }
+
+    dispatch(
+      pushModal({
+        modal: {
+          name: 'confirm-delete',
+          options: {
+            message: t('Are you sure you want to delete this rule?'),
+            onConfirm: async () => {
+              try {
+                await send('rule-delete', id);
```

Comment:
It would be nice to have this function be a redux action, we can consider creating a redux slice for rules in the future